src/main/pages/workspace-admin/recipes.adoc

```diff
@@ -11,42 +11,56 @@ folder: workspace-admin
+The Docker-formatted container image recipe pulls an image from a Docker registry or uses the local image.  The recipe then runs the image and creates a pod that references this image in the container specification. The following are Docker-formatted container image requirements and limitations for a workspace machine:
 
-1.  Non-terminating `CMD` or `ENTRYPOINT`. If you use your custom image, make sure it has something like `tail -f /dev/null` as one of its main processes.
-2.  No processes and operations with `sudo` in `CMD` (OpenShift only, see link:openshift-config.html#enable-ssh-and-sudo[Enable sudo])
-3.  OpenShift only - it is recommended to use https://github.com/eclipse/che-dockerfiles/tree/master/recipes/stack-base[Che base stacks] that are certified to run OK on OpenShift. You can also build your own image, but inherit from one of base stacks.
+1.  Use a non-terminating `CMD` or `ENTRYPOINT`.  You can use `tail -f /dev/null` as one of the main processes if you are using a custom image.
+2.  For OpenShift only: 
+ 
+* Do not use any processes and operations with `sudo` in `CMD`. See link:openshift-config.html#enable-ssh-and-sudo[Enable SSH and `sudo`] for more information.
 
-[id="dockerfile"]
-== Dockerfile
+* Use https://github.com/eclipse/che-dockerfiles/tree/master/recipes/stack-base[Che base stacks]. You can also build your own image, but inherit from one of the base stacks.
 
-A Dockerfile is a set of instructions that Docker performs to build an image. After you provide a Dockerfile for your workspace machine, Che will initiate Docker build, and once successful, run the resulting image. There are certain limitations here as well:
+[id="dockerfile-definition-and-limitations"]
+== Dockerfile definition and limitations
 
-1.  There’s no context in docker build in Che. It means that commands like `COPY` and `ADD` will fail.
-2.  Build of huge images may take much time, so, if you have a long Dockerfile, it’s better to build an image locally, push it to DockerHub and then use it as DockerImage recipe type (see above). Che has a reasonable timeout for a workspace to start which is 5 mins.
+A Dockerfile is a set of instructions that Docker performs to build an image. After you provide a Dockerfile for your workspace machine, Che initiates a Docker build and runs the resulting image. The following are the limitations:
 
-[id="composefile"]
-== Composefile
+1.  The `COPY` and `ADD` instructions fail because there is no context in `docker build`.
+2.  Large images can take a long time to build.  If you have a long Dockerfile, it is recommended to build that image locally, push it to DockerHub and then use the pushed image as a Docker-formatted container image recipe type. The start timeout for a workspace is five minutes.
```

Comment:
* adoc mark-up
* no 'if'
* serial comma

```suggestion
. To avoid long build times with long Dockerfiles, build your image locally, push it to DockerHub, and then use the pushed image as a Docker-formatted container image recipe type. The start timeout for a workspace is five minutes.
```